pages/api/v1/progressReport/addReportRating.ts

```diff
@@ -52,15 +52,23 @@ const handler: NextApiHandler<{ message: string; progress_report?: object }> = a
 		}
 
 		const postData = postDoc.data() as Post;
-		const progressReport = postData.progress_report || {};
-		let ratings = progressReport.ratings || [];
-
-		ratings = ratings.filter((r: { user_id: string; rating: number }) => r.user_id !== user_id);
-		ratings.push({ rating, user_id });
-		progressReport.ratings = ratings;
+		const progressReports = postData.progress_report || [];
+
+		const updatedProgressReports = progressReports.map((report: IProgressReport) => {
+			if (report.id === reportId) {
+				let ratings = report.ratings || [];
+				ratings = ratings.filter((r: { user_id: string }) => r.user_id !== user_id);
+				ratings.push({ rating, user_id });
+				return {
+					...report,
```

Comment:
Please use created_at.toDate() for report.created_at otherwise it will save the firebase timestamp in different format